numpy/core/arrayprint.py

```diff
@@ -949,7 +972,7 @@ def fillFormat(self, data):
         if len(finite_vals) == 0:
             self.pad_left = 0
             self.pad_right = 0
-            self.trim = '.'
+            self.trim = '.'  # do not override
```

Comment:
The comment is confusing since the code is overriding the value

Reply:
I figured out that I was referring to the case of 'g' where I set `trim='k'` but it is wrong. I removed those changes and added a note for implementing correctly the general format 'g'.